RedditSharp/Reddit.cs

```diff
@@ -275,24 +276,43 @@ public Domain GetDomain(string domain)
             return new Domain(this, uri, WebAgent);
         }
 
-        public JToken GetToken(Uri uri)
-        {
-            var url = uri.AbsoluteUri;
-
-            if (url.EndsWith("/"))
-                url = url.Remove(url.Length - 1);
-
-            var request = WebAgent.CreateGet(string.Format(GetPostUrl, url));
-            var response = request.GetResponse();
-            var data = WebAgent.GetResponseString(response.GetResponseStream());
-            var json = JToken.Parse(data);
-
-            return json[0]["data"]["children"].First;
+        public JToken GetToken(Uri uri)
+        {
+            return GetToken(uri.AbsoluteUri);
+        }
+
+        public JToken GetToken(string url)

```

Comment:
I don't think this is necessary, better to just leave GetToken(Uri) implementation, same goes for GetPost